observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerDeploymentProcessor.java

```diff
@@ -25,38 +26,34 @@
 import org.jboss.as.weld.WeldCapability;
 import org.wildfly.extension.micrometer.api.MicrometerCdiExtension;
 import org.wildfly.extension.micrometer.metrics.MetricRegistration;
-import org.wildfly.extension.micrometer.metrics.MicrometerCollector;
 import org.wildfly.subsystem.service.ServiceDependency;
 import org.wildfly.subsystem.service.ServiceInstaller;
 
 class MicrometerDeploymentProcessor implements DeploymentUnitProcessor {
     static final String WELD_CAPABILITY_NAME = "org.wildfly.weld";
+    static final AttachmentKey<MicrometerService> CONFIG_ATTACHMENT_KEY = AttachmentKey.create(MicrometerService.class);
 
-    private final MicrometerSubsystemRegistrar.MicrometerDeploymentConfiguration config;
-
-    MicrometerDeploymentProcessor(MicrometerSubsystemRegistrar.MicrometerDeploymentConfiguration config) {
-        this.config = config;
+    MicrometerDeploymentProcessor() {
     }
 
     @Override
-    public void deploy(DeploymentPhaseContext deploymentPhaseContext) throws DeploymentUnitProcessingException {
-        final DeploymentUnit deploymentUnit = deploymentPhaseContext.getDeploymentUnit();
+    public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitProcessingException {
+        final DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
 
-        ServiceDependency<MicrometerCollector> collector = ServiceDependency.on(MICROMETER_COLLECTOR_RUNTIME_CAPABILITY.getCapabilityServiceName());
-        Supplier<MetricRegistration> factory = () -> collector.get().collectResourceMetrics(
+        ServiceDependency<MicrometerService> serviceDependency = ServiceDependency.on(MICROMETER_SERVICE_SERVICE_NAME);
+        Supplier<MetricRegistration> factory = () -> serviceDependency.get().collectResourceMetrics(
                 deploymentUnit.getAttachment(DeploymentModelUtils.DEPLOYMENT_RESOURCE),
                 deploymentUnit.getAttachment(DeploymentModelUtils.MUTABLE_REGISTRATION_ATTACHMENT),
-                createDeploymentAddressPrefix(deploymentUnit)::append,
-                this.config.getSubsystemFilter());
+                createDeploymentAddressPrefix(deploymentUnit)::append);
         ServiceInstaller.builder(factory)
                 .requires(ServiceDependency.on(DeploymentCompleteServiceProcessor.serviceName(deploymentUnit.getServiceName())))
-                .requires(collector)
+                .requires(serviceDependency)
                 .asActive()
                 .onStop(MetricRegistration::unregister)
                 .build()
-                .install(deploymentPhaseContext);
+                .install(phaseContext);
 
-        registerCdiExtension(deploymentPhaseContext);
+        registerCdiExtension(phaseContext, serviceDependency.get());
```

Comment:
The missing small comment:

s/serviceDependency.get()/deploymentUnit.getAttachment(CONFIG_ATTACHMENT_KEY)/g

The serviceDependency.get() will work because MicrometerDependencyProcessor has established a situation where this DUP won't even run until that service is started. But it's clearer for maintainers / less fragile if the code here explicitly gets the value via the directly-established mechanism, the attachment.

Reply:
Changed